### PR TITLE
Adjust zombie solider drops

### PR DESCRIPTION
**Pull request by mugling**
- Drop guns from `guns_pistol_milspec`, `guns_rifle_milspec` and `guns_shotgun_milspec`, heavily biased towards the first two groups (78d80cf, b469125, f517ff2)
- Always drop guns with magazines and sometimes (40%) with ammo (5a78da2, a262ed2)
- Bias against winter clothing (35%) in preference of standard (65%)
- Increase chance for `knife_combat` which should otherwise become rarer (d1758bb, 59f0b33)
- Increase chance to carry `40mm` grenades which are otherwise uncraftable (526b1b4)

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Might also want to consider the drops I added in here:

#14755

**Comment by mugling** [comment]
> Might also want to consider the drops I added in here

I've just read through that PR now.

I've switched to using `guns_*_milspec` so that we can easily adjust relative gun rarity when new items are added and/or removed.

Overall this PR reduces the ammo dropped so I can't see much controversy there. I did increase the frequency of the medic zombie from 2% to 5%.

I'm not sure how the increase in `40mm` ammo will be received. It's currently uncraftable (and should remain so) therefore a reasonably frequent source seems reasonable from a balance pov?

**Comment by mugling** [comment]
We could also consider adding some ammo pouches?

**Comment by Coolthulhu** [comment]
> I did increase the frequency of the medic zombie from 2% to 5%.

Did you nerf the drops? The problem with medics is that they often carry a bit of everything at once - morphine, antibiotics, first aid, guaranteed bandages, tramadol and usual drops on top of that.

Why remove knife_rambo from non-combat drops? This is a survival knife, a tool rather than a weapon (it has really bad combat stats).

**Comment by mugling** [comment]
> Why remove knife_rambo from non-combat drops?

I've replaced it in the `camping and `allsporting` itemgroups (a931ec1)

**Comment by mugling** [comment]
> The problem with medics is that they often carry a bit of everything at once

As of 6e10ca8 you get one drop out of `soldier_meds` per zombie solider medic

**Comment by Mecares** [comment]
Instead of always having magazines i would prefer it if they would be having 0-3 but on average one and not all guns having always a magazine in it because being overwhelmed is likely to happen when you are busing reloading instead of shooting the Zeds.

**Comment by mugling** [comment]
@Mecares can you rephrase that I don't follow?

**Comment by Mecares** [comment]
Okay, sorry was writing in a hurry. I like the idea of more frequent magazine spawns in soldier zombies. But i would prefer if it would not be guaranteed  but a bit randomly distributed. Let not every weapon spawn with a magazine but to even it out let some Zeds have a second or third one. My thought was that the likely moment a soldier gets overwhelmed bye zeds is when he either ran out of ammo or during the process of reloading making it plausible for some weapons to not contain magazines.

**Comment by mugling** [comment]
@Mecares whilst that's not a bad idea unfortunately the current `item_group` code couldn't support (in any practical sense) guns with a variable number of **matching** magazines

**Comment by Mecares** [comment]
Okay in that case better discard that spontaneous thought of mine the whole magazine Implementation seems like already a huge amount of work without that making  it more complicated.
